### ISSUE TITLE
v3.5.1 Regression: Permissions problems in Module Manager

### ISSUE THREAD

**Issue by severdia** [joomla/joomla-cms#9856]
#### Steps to reproduce the issue
Open the Module Manager and some or all of the modules are "disabled".

#### Expected result
All modules should be editable and toggle publish state as shown below.

![screen shot 2016-04-11 at 11 58 23 am](https://cloud.githubusercontent.com/assets/355924/14439127/6fe02b64-ffde-11e5-8a41-2864a444def8.png)

#### Actual result
Sometimes all modules and sometimes only unpublished modules are being incorrectly identified as disabled modules. 

![screen shot 2016-04-11 at 11 59 45 am](https://cloud.githubusercontent.com/assets/355924/14439144/89646762-ffde-11e5-99e6-f5e936a8263a.png)

#### System information (as much as possible)

Database Version	5.5.31-30.3-log
Database Collation	utf8_general_ci
Database Connection Collation	utf8mb4_general_ci
PHP Version	5.6.19
Web Server	Apache/2.2.24 (Unix) mod_hive/5.0 mod_ssl/2.2.24 OpenSSL/1.0.0-fips mod_auth_passthrough/2.1 mod_bwlimited/1.4 mod_fastcgi/2.4.6 mod_fcgid/2.3.6
WebServer to PHP Interface	cgi-fcgi
Joomla! Version	Joomla! 3.5.1 Stable [ Unicorn ] 05-April-2016 22:45 GMT
Joomla! Platform Version	Joomla Platform 13.1.0 Stable [ Curiosity ] 24-Apr-2013 00:00 GMT

#### Additional comments
There appears to be a problem with the check for module state. I did a diff on this file:

/administrator/components/com_modules/views/modules/tmpl/default.php

And I disabled the sections where the check is done. See the attached file for a working file, but this is clearly not the best solution. It was just to get the Module Manager working again.

[default.php.zip](https://github.com/joomla/joomla-cms/files/213716/default.php.zip)

I reverted these two places to the previous Joomla version:

(line 117)
```
<?php if ($item->enabled > 0) : ?>
<?php echo JHtml::_('grid.id', $i, $item->id); ?>
<?php endif; ?>
```

(line 121)
```
<?php // Check if extension is enabled ?>
<?php if ($item->enabled > 0) : ?>
<?php echo JHtml::_('jgrid.published', $item->published, $i, 'modules.', $canChange, 'cb', $item->publish_up, $item->publish_down); ?>
<?php // Create dropdown items ?>
<?php JHtml::_('actionsdropdown.duplicate', 'cb' . $i, 'modules'); ?>
<?php $action = $trashed ? 'untrash' : 'trash'; ?>
<?php JHtml::_('actionsdropdown.' . $action, 'cb' . $i, 'modules'); ?>
<?php // Render dropdown list ?>
<?php echo JHtml::_('actionsdropdown.render', $this->escape($item->title)); ?>
<?php else : ?>
<?php // Extension is not enabled, show a message that indicates this. ?>
<button class="btn-micro hasTooltip" title="<?php echo JText::_('COM_MODULES_MSG_MANAGE_EXTENSION_DISABLED'); ?>"><i class="icon-ban-circle"></i></button>
<?php endif; ?>
```

**Comment by brianteeman** [comment]
Very strange as we havent had a single other report in such  fundamental
part of joomla. Are you able to replicate this on a completely clean
install of joomla with NO extensions

On 11 April 2016 at 20:19, Ron Severdia <notifications@github.com> wrote:

> Steps to reproduce the issue
>
> Open the Module Manager and some or all of the modules are "disabled".
> Expected result
>
> All modules should be editable and toggle publish state as shown below.
>
> [image: screen shot 2016-04-11 at 11 58 23 am]
> <https://cloud.githubusercontent.com/assets/355924/14439127/6fe02b64-ffde-11e5-8a41-2864a444def8.png>
> Actual result
>
> Sometimes all modules and sometimes only unpublished modules are being
> incorrectly identified as disabled modules.
>
> [image: screen shot 2016-04-11 at 11 59 45 am]
> <https://cloud.githubusercontent.com/assets/355924/14439144/89646762-ffde-11e5-99e6-f5e936a8263a.png>
> System information (as much as possible)
>
> Database Version 5.5.31-30.3-log
> Database Collation utf8_general_ci
> Database Connection Collation utf8mb4_general_ci
> PHP Version 5.6.19
> Web Server Apache/2.2.24 (Unix) mod_hive/5.0 mod_ssl/2.2.24
> OpenSSL/1.0.0-fips mod_auth_passthrough/2.1 mod_bwlimited/1.4
> mod_fastcgi/2.4.6 mod_fcgid/2.3.6
> WebServer to PHP Interface cgi-fcgi
> Joomla! Version Joomla! 3.5.1 Stable [ Unicorn ] 05-April-2016 22:45 GMT
> Joomla! Platform Version Joomla Platform 13.1.0 Stable [ Curiosity ]
> 24-Apr-2013 00:00 GMT
> Additional comments
>
> There appears to be a problem with the check for module state. I did a
> diff on this file:
>
> /administrator/components/com_modules/views/modules/tmpl/default.php
>
> And I disabled the sections where the check is done. See the attached file
> for a working file, but this is clearly not the best solution. It was just
> to get the Module Manager working again.
>
> default.php.zip
> <https://github.com/joomla/joomla-cms/files/213716/default.php.zip>
>
> I reverted these two places to the previous Joomla version:
>
> (line 117)
>
> <?php if ($item->enabled > 0) : ?>
> <?php echo JHtml::_('grid.id', $i, $item->id); ?>
> <?php endif; ?>
>
> (line 121)
>
> <?php // Check if extension is enabled ?>
> <?php if ($item->enabled > 0) : ?>
> <?php echo JHtml::_('jgrid.published', $item->published, $i, 'modules.', $canChange, 'cb', $item->publish_up, $item->publish_down); ?>
> <?php // Create dropdown items ?>
> <?php JHtml::_('actionsdropdown.duplicate', 'cb' . $i, 'modules'); ?>
> <?php $action = $trashed ? 'untrash' : 'trash'; ?>
> <?php JHtml::_('actionsdropdown.' . $action, 'cb' . $i, 'modules'); ?>
> <?php // Render dropdown list ?>
> <?php echo JHtml::_('actionsdropdown.render', $this->escape($item->title)); ?>
> <?php else : ?>
> <?php // Extension is not enabled, show a message that indicates this. ?>
> <button class="btn-micro hasTooltip" title="<?php echo JText::_('COM_MODULES_MSG_MANAGE_EXTENSION_DISABLED'); ?>"><i class="icon-ban-circle"></i></button>
> <?php endif; ?>
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/joomla/joomla-cms/issues/9856>
>



-- 
Brian Teeman
Co-founder Joomla! and OpenSourceMatters Inc.
http://brian.teeman.net/

**Comment by severdia** [comment]
One site is clean (upgraded from 3.4.8 using the Joomla installer) and another has some extensions installed.

**Comment by brianteeman** [comment]
In extensions->manage->manage
Do you have anything disabled

On 11 April 2016 at 20:35, Ron Severdia <notifications@github.com> wrote:

> One site is clean (upgraded from 3.4.8 using the Joomla installer) and
> another has some extensions installed.
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/joomla/joomla-cms/issues/9856#issuecomment-208519673>
>



-- 
Brian Teeman
Co-founder Joomla! and OpenSourceMatters Inc.
http://brian.teeman.net/

**Comment by severdia** [comment]
Yes. The following default extensions:

Authentication - Gmail
Authentication - LDAP
Captcha - ReCaptcha
Content - Smart Search
Installer
Search - Tags
System - Language Code
Two Factor Authentication
Two Factor Authentication - YubiKey
User - Contact Creator	
User - Profile
Weblinks		
Weblinks Extension Package

**Comment by brianteeman** [comment]
Are you sure that Installer is disabled - thats a protected extension that
you should not be able to change the status of

On 11 April 2016 at 20:57, Ron Severdia <notifications@github.com> wrote:

> Yes. The following default extensions:
>
> Authentication - Gmail
> Authentication - LDAP
> Captcha - ReCaptcha
> Content - Smart Search
> Installer
> Search - Tags
> System - Language Code
> Two Factor Authentication
> Two Factor Authentication - YubiKey
> User - Contact Creator
>
> User - Profile
> Weblinks
>
> Weblinks Extension Package
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/joomla/joomla-cms/issues/9856#issuecomment-208530133>
>



-- 
Brian Teeman
Co-founder Joomla! and OpenSourceMatters Inc.
http://brian.teeman.net/

**Comment by severdia** [comment]
Yes.

**Comment by brianteeman** [comment]
The only way that that extension can be disabled is if you have manually edited it in the database or if you have a corrupt database.  Even then when I manually edit my database I cannot replicate your issue. It all points to an issue with your specific web site and nothing wrong with the core.

**Comment by severdia** [comment]
Not true. That's actually the installer for the RocketTheme template. It's not a native Joomla extension.

**Comment by severdia** [comment]
I should add that it was working find in 3.4.8 before the update to 3.5.1. A diff on the database shows almost nothing changed related to this issue (like the extensions table) and disabling this check in the view file fixes the issue. It's definitely a Joomla issue.

**Comment by brianteeman** [comment]
>  That's actually the installer for the RocketTheme template. It's not a native Joomla extension. 

I couldnt guess that as 1. there is a core component called Installer and 2. you said there were no extensions installed.

If you can repeat this with a CLEAN install of Joomla with NOTHING installed then I will re open this

**Comment by severdia** [comment]
So it's either an application bug or an upgrade bug. Either way it's a bug that I can easily reproduce. Closing this issue doesn't make it go away.

**Comment by brianteeman** [comment]
Or a bug in something you have done to your own site. Unless it can be
replicated I have to assume the latter. It doesn't help when you provide
inaccurate information.

**Comment by rdeutz** [comment]
The error message says that a property doesn't exists, this property come from the database so when the modules table has the published column it must be something on the way, can you var_dump $item in the foreach, that can be a hint what is wrong. 

I would try to use a backup and update again.

**Comment by stevejburge** [comment]
I'm able to confirm this bug after working with a customer.

It was a Joomla 3.5.1 site, hosted with Siteground. We installed Advanced Module Manager as a temporary workaround.

My guess it that Siteground's PHP version or configuration may have an impact. @severdia were your sites hosted there?<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/9856">issues.joomla.org/joomla-cms/9856</a>.</sub>

**Comment by brianteeman** [comment]
All my own sites are there as are the joomla.com and demojoomla.org sites
so I doubt very much it is hosting related.

**Comment by severdia** [comment]
Yes, it's hosted at Siteground. I just now reverted the default.php file back to the 3.5.1 "broken" version and the problem no longer exists (even with maximum reporting). That could confirm Siteground changed their config to fix it (at least on my server) because my site hasn't been touched in any way since I reported this.

**Comment by stevejburge** [comment]
Interesting. This particular customer did report that the problem started and stopped as Siteground moved them between different PHP versions and configurations.

**Comment by alexblanc** [comment]
Am sorry to report, but...... 

Siteground PHP variations are not the issue -  we're experiencing the exact same mysterious quandary. 

We have two sites on the same managed server (not Siteground) - no PHP settings have been changed whatsoever (we manage the serve;-).

After upgrading from 3.4.8 to 3.5.2, one of the sites shows Modules perfectly, the other has ALL Modules listed as Disabled, though they are not. (In extensions->manage->manage)  Interestingly, they still work. (show/hide title, etc)

We've tried replacing the entire administrator/components/com_modules folder with a fresh copy, to no avail.

Both sites use YooTheme templates, and have the same 'basic' extensions / latest versions installed (akeebabackup, admintools, jceditor & breezingforms).  

The site with the Module Issue also has AcyMailing Enterprise(latest), and uses Widgetkit 1.x as opposed to the working site using Widgetkit 2.x

Only other difference we see is that Weblinks is installed in site with the Issue, but missing from the working site.

Any clues, anyone?  <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/9856">issues.joomla.org/joomla-cms/9856</a>.</sub>
